textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/Span.java

```diff
@@ -3,22 +3,40 @@
 public class Span {
     //The name of the field (in the tuple) where this span is present
     private String fieldName;
-    //The start of the span
+    //The start of the span. It is the position of the first character of span in the document.
     private int start;
-    //The end of the span
+    //The end of the span.It is the position of the first character of span in the document
     private int end;
     //The key we are searching for eg: regex
     private String key;
     //The value matching the key
     private String value;
-    
-    
-    public Span(String fieldName, int start, int end, String key, String value) {
+    // The token position of the span
+    private int tokenOffset;
+
+    /*
+    Example:
+        Value = "The quick brown fox jumps over the lazy dog"
+        Now the Span for brown should be
+        start = 10 : position of character 'b'
+        end = 15 : position of character 'n'

```

Comment:
Also you are setting few properties while creating the field

org.apache.lucene.document.FieldType luceneFieldType = new org.apache.lucene.document.FieldType();
                luceneFieldType.setIndexOptions( IndexOptions.DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS );
                luceneFieldType.setStored(true);
                luceneFieldType.setStoreTermVectors( true );
                luceneFieldType.setStoreTermVectorOffsets( true );
                luceneFieldType.setStoreTermVectorPayloads( true );
                luceneFieldType.setStoreTermVectorPositions( true );
                luceneFieldType.setTokenized( true );